Test/Core/Cloning/SpecificCloningTest.cs

```diff
@@ -40,25 +43,53 @@ public class SpecificCloningTest
 			Assert.AreEqual(source.Options, target.Options);
 			Assert.AreEqual(source.MatchTimeout, target.MatchTimeout);
 		}
-		[Test] public void CloneCultureInfo()
+		[Test]
+		public void CloneCultureInfo()
 		{
 			CultureInfo source = new CultureInfo("en-US");
 			CultureInfo target = source.DeepClone();
 
 			Assert.AreNotSame(source, target);
 			Assert.AreEqual(source, target);
 		}
-		[Test] public void CloneHashSet()
+		[Test]
+		public void CloneHashSet_ValueType()
 		{
 			Random rnd = new Random();
-			HashSet<int> source = new HashSet<int>(Enumerable.Range(0, 50).Select(i => rnd.Next()));
+			var values = Enumerable.Range(0, 50).Select(i => rnd.Next());
+			HashSet<int> source = new HashSet<int>(values);
 			HashSet<int> target = source.DeepClone();
 
 			Assert.AreNotSame(source, target);
-			CollectionAssert.AreEquivalent(source, target);
+			CollectionAssert.AreEqual(source, target);
 		}
 
-		[Test] public void CloneContentRef()
+		[Test]
+		public void CloneHashSet_ReferenceType()
+		{
+			var rnd = new Random();
+			var values = Enumerable.Range(0, 50).Select(i => new TestObject(rnd, 0));
+			HashSet<TestObject> source = new HashSet<TestObject>(values);
+			HashSet<TestObject> target = source.DeepClone();
+
+			Assert.AreNotSame(source, target);
+			CollectionAssert.AreEqual(source, target);
+		}
+
+		[Test]
+		public void CloneHashSet_ReferenceAndNulls()
+		{
+			var rnd = new Random();
+			var values = Enumerable.Range(0, 50).Select(i => i % 2 == 0 ? null : new TestObject(rnd, 0));
```

Comment:
This range contains 50% null values, but the hashset will only contain one. Maybe adjust the code to reflect that?